dspace-api/src/main/java/org/dspace/importer/external/service/DoiCheck.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.importer.external.service;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * Utility class that provides methods to check if a given string is a DOI and exists on CrossRef services
+ *
+ * @author Corrado Lombardi (corrado.lombardi at 4science.it)
+ */
+public class DoiCheck {
```

Comment:
If we want to start by updating this PR **first** (since it's nearly working), then I'd appreciate it if you could add Unit Tests to this class as described in the discussion at https://github.com/DSpace/DSpace/pull/8236#discussion_r857815979

As noted in that discussion, the DOI URIs here also might need to be made configurable, though the defaults you have here are fine.